sprockets/mixins/metrics/statsd.py

```diff
@@ -110,13 +113,42 @@ class StatsDCollector(object):
     METRIC_TYPES = {'c': 'counters',
                     'ms': 'timers'}
 
-    def __init__(self, host, port, namespace='sprockets',
+    def __init__(self, host, port, proto='udp', namespace='sprockets',
                  prepend_metric_type=True):
         self._host = host
         self._port = int(port)
+        self._address = (self._host, self._port)
         self._namespace = namespace
         self._prepend_metric_type = prepend_metric_type
-        self._sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
+
+        if proto == 'udp':
+            self._tcp = False
+            self._sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
+        else:
```

Comment:
Explicitly check for `tcp`. Otherwise, any value other than `udp` will result in `tcp`.